modeling/gbert_classifier.py

```diff
@@ -224,52 +260,112 @@ def make_model(label):
         logger.info(f'Epoch {epoch + 1}/{EPOCHS}')
         logger.info('-' * 10)
-        train_f1, train_loss = train_epoch(
+        train_fbeta, train_metrics, train_params, train_loss = train_epoch(
             model,
             train_data_loader,
             loss_fn,
             optimizer,
             device,
             scheduler,
-            len(df_train)
         )
-        logger.info(f'Train loss {train_loss} F1 {train_f1}')
-        val_f1, val_loss = eval_model(
+        logger.info(f'Train loss {train_loss} Fbeta {train_fbeta}')
+        val_fbeta, val_metrics, val_params,  val_loss = eval_model(
             model,
             val_data_loader,
             loss_fn,
             device,
-            len(df_val)
         )
-        logger.info(f'Val   loss {val_loss} F1 {val_f1}')
+        logger.info(f'Val   loss {val_loss} Fbeta {val_fbeta}')
         print()
-        history['train_f1'].append(train_f1)
+        history['train_fbeta'].append(train_fbeta)
         history['train_loss'].append(train_loss)
-        history['val_f1'].append(val_f1)
+        history['val_fbeta'].append(val_fbeta)
         history['val_loss'].append(val_loss)
-        if val_f1 > best_f1:
-            file_name = f"./models/model_gbert_pool_{label}_{t}.bin"
+        if val_fbeta > best_fbeta:
+            file_name = f"./models/model_gbertbase_{label}_{t}.bin"
             torch.save(model.state_dict(), file_name)
-            best_f1 = val_f1
+            for k,v in val_metrics.items():
+                mlflow_logger.add_metric(k, v)
+            for k,v in val_params.items():
+                mlflow_logger.add_param(k, v)
+            for k,v in train_metrics.items():
+                mlflow_logger.add_metric(k, v)
+            for k,v in train_params.items():
+                mlflow_logger.add_param(k, v)
+            best_fbeta = val_fbeta
+
+    #############################################
+    #MLflow logging
+
+    constant_params = {
+                    'epochs': EPOCHS,
+                    'batch_size': BATCH_SIZE,
+                    'max_len': MAX_LEN,
+                }
+    mlflow_logger.add_tag("cycle4", True)
+    mlflow_logger.add_param("normalization", 'norm')
+    mlflow_logger.add_param("vectorizer", 'deepset/gbert-base')
+    mlflow_logger.add_param("model", "deepset/gbert-base")
+    # I'm re-using the grid_search_params field in order to no open too many mlflow columns
+    mlflow_logger.add_param("grid_search_params", str(constant_params)[:249]) 
+    mlflow_logger.add_param("lr", learning_rate)
+    mlflow_logger.add_param('pos_weight', positive_class_weight)
+
+    mlflow_logger.add_param("saved_model", f"model_gbertbase_{label}_{t}")
+    mlflow_logger.add_param("label", data.current_label)
+    mlflow_logger.add_param("balance_method", data.balance_method)
+    if data.balance_method:
+        mlflow_logger.add_param("sampling_strategy", data.sampling_strategy)
+    mlflow_logger.add_model(None)
```

Comment:
FYI: As you never set a model, you should not need to do this. It is by default set to None and therefore, not logging.
But it might be better to leave it in here, to have it explicit.